src/collection.jl

```diff
@@ -0,0 +1,100 @@
+# helper function to match extension
+# this will be removed when there is a compat entry of endswith with regex compatibility
+function match_extension(str::String, ext::Regex)
+    r = deepcopy(ext)
+    if !endswith(r.pattern, "\$")
+        r = Regex(r.pattern * "\$", r.compile_options, r.match_options)
+    end
+    return occursin(r, str)
+end
+
+match_extension(str::String, ext::String) = endswith(str, ext)
```

Comment:
```suggestion
match_extension(str::String, ext) = endswith(str, ext)
```

I would avoid specializing the extension in this version